pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -160,28 +163,45 @@ func (rq *ResourceQuotaController) enqueueResourceQuota(obj interface{}) {
 	rq.queue.Add(key)
 }
 
+func (rq *ResourceQuotaController) addQuota(obj interface{}) {
+	key, err := controller.KeyFunc(obj)
+	if err != nil {
+		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+
+	resourceQuota := obj.(*api.ResourceQuota)
+	if len(resourceQuota.Status.Used) == 0 {
```

Comment:
how about the following:

```
// if we declared an intent that is not yet captured in status (prioritize it)
if len(resourceQuota.Spec.Hard) != len(resourceQuota.Status.Hard) {
   rq.missingUsageQueue.Add(key)
}

// if we declared a constraint that has no usage (which this controller can calculate, prioritize it)
for constraint := range resourceQuota.status.Hard {
  if _, usageFound := resourceQuota.Status.Used[constraint]; !usageFound {
    for evaluator := range evaluators {
       if intersection := Intersection(evaluator.MatchesResources(), matchedResources); len(intersection) != 0 {
       rq.missingUsageQueue.Add(key)
       return
       }
    }
  }
}

// no special priority, go in normal recalc queue
rq.queue.Add(key)
```

we could then also call this same addQuota func from Add and UpdateFunc in controller def.

thoughts?